### ISSUE TITLE
The meaning of the `grad` on the `LieTensor.SO3`

### ISSUE THREAD

**Issue by zeroAska** [pypose/pypose#257]
### 📚 The doc issue

After performing autograd on a LieTensor SO3 tensor `T`, is the resulting `T.grad` always a LieTensor as well? When the optimizer apply the update to the LieTensor, is the update operation on the manifold?

I ran into such a situation after running `loss.backward()`:
```
(Pdb) p T_init
SO3Type LieTensor:
LieTensor([[-1.4035e-03, -1.2569e-04,  1.1781e-02,  9.9993e-01]],
          device='cuda:0', requires_grad=True)
(Pdb) p T_init.grad
tensor([[-1.3532, -0.0474, -0.4073,  0.0000]], device='cuda:0')
```
Where the `T_init.grad`'s data doesn't look like a quaternion any more. Is this value directly added on the `T_init`'s data, or is it converted to a quaternion first (something like `SO3(T_init.grad)`) then multipled to the `T_init`?

**Comment by wang-chen** [comment]
You need to call `LieTensor.add` to update lietensor by grad. Here is more [detailed mathematical definition](https://pypose.org/docs/main/generated/pypose.add/#pypose.add) for gradient of Lie Groups
